apis_core/generic/views.py

```diff
@@ -104,14 +105,38 @@ def get_table_class(self):
 
     def get_table_kwargs(self):
         kwargs = super().get_table_kwargs()
-        columns = self.request.GET.getlist("columns", [])
-        column_fields = [
-            field for field in self.model._meta.fields if field.name in columns
+
+        # we look at the selected columns and exclude
+        # all modelfields that are not part of that list
+        selected_columns = self.request.GET.getlist(
+            "columns",
+            self.get_filterset(self.get_filterset_class()).form["columns"].initial,
+        )
+        modelfields = self.model._meta.fields
```

Comment:
The current solution doesnt allow to deselect all columns. If something in the form is changed the whole form is put to the GET params. So to see if we need to use the initial values or the empty columns list we check whether the GET params are empty or not.

```suggestion
        selected_columns = self.request.GET.getlist("columns") if len(
            self.request.GET) > 0 else self.get_filterset(self.get_filterset_class()).form["columns"].initial
```